### PR TITLE
Rewards flow

### PR DESCRIPTION
**Pull request by oetyng**
Enables the initiation of reward payouts, through various changes such as:
- Refactor to have consistent elder state between elder transitions.
- Waiting for a threshold of elders in genesis section, before genesis section actor is created.
- ... and other fixes and tweaks.

Todo: Completion of an initiated reward payout.

### CODE REVIEW

**Comment by happybeing** [comment]
If it's relevant to this PR, please can you include `info!()` log messages for reward received (ideally including wallet balance in that or a co-incident log message) so I can add this to `vdash`. Very excited to see this!

I had a quick scan of the PR so far and noticed the following, which I'm including here just for my own record:
```
// Rewards log messages 28-01-21

//// Node ////

info!("Registering wallet: {}", wallet);
//  src/node/node_duties/mod.rs

//// Section ////

info!("Genesis balance: {}", balance)
// src/node/elder_duties/key_section/transfers/mod.rs

info!("Section Actor balance: {}", self.actor.balance());
// src/node/elder_duties/data_section/rewards/section_funds.rs
```

**Comment by oetyng** [comment]
> If it's relevant to this PR, please can you include `info!()` log messages for reward received (ideally including wallet balance in that or a co-incident log message) so I can add this to `vdash`. Very excited to see this!

Hey Mark!

Logs for rewards received could absolutely be added, but that would probably be when a node queries its replicas for new incoming transfers, and examines the message of any credits.
Will look at it for upcoming work!

File: src/node/elder_duties/data_section/rewards/mod.rs [hunk 1]
```diff
@@ -91,15 +93,16 @@ impl Rewards {
     /// Issues a query to existing Replicas
     /// asking for their events, as to catch up and
     /// start working properly in the group.
-    #[allow(unused)]
     pub async fn catchup_with_replicas(&self) -> Result<NodeOperation> {
         info!("Rewards: Catching up with our Replicas (section actor history)!");
         // prepare actor init
-        let pub_key = self.section_funds.replicas();
+        let pub_key = self.section_funds.replicas(); // this can't be right?
```
Comment (joshuef) [inline src/node/elder_duties/data_section/rewards/mod.rs:99]: 🤔 is it?

File: src/node/elder_duties/data_section/rewards/mod.rs [hunk 2]
```diff
@@ -110,10 +113,19 @@ impl Rewards {
...
+    pub async fn init_transition(&mut self, elder_state: ElderState) -> Result<NodeOperation> {
+        self.section_funds
+            .init_transition(elder_state)
+            .await
+            .convert()
     }
 
+    // /// After Elder change, we transition to a new
+    // /// transfer actor, as there is now a new keypair for it.
+    // pub async fn transition(&mut self, to: TransferActor<Validator>) -> Result<NodeOperation> {
```
Comment (joshuef) [inline src/node/elder_duties/data_section/rewards/mod.rs:125]: lets remove this commented code

File: src/node/node_ops.rs [hunk 3]
```diff
@@ -117,9 +118,41 @@ pub enum NodeDuty {
...
+    /// of various places using the multisig of the section.
+    InitiateElderChange {
+        /// The prefix of our section.
+        prefix: Prefix,
+        /// The BLS public key of our section.
+        key: PublicKey,
+        /// The set of elders of our section.
+        elders: BTreeSet<XorName>,
+    },
+    ///
```
Comment (joshuef) [inline src/node/node_ops.rs:149]: missing docs

File: src/node/node_ops.rs [hunk 4]
```diff
@@ -117,9 +118,41 @@ pub enum NodeDuty {
...
+    FinishElderChange {
+        previous_key: PublicKey,
+        new_key: PublicKey,
+    },
+    ///
```
Comment (joshuef) [inline src/node/node_ops.rs:154]: missing docs

File: src/utils.rs [hunk 5]
```diff
@@ -43,8 +43,8 @@ pub(crate) fn new_auto_dump_db<D: AsRef<Path>, N: AsRef<Path>>(
     if init_mode == Init::New {
         debug!("Creating auto dump database at {}", db_path.display());
         fs::create_dir_all(db_dir)?;
-        let db = PickleDb::new_bin(db_path, PickleDbDumpPolicy::AutoDump);
-        return Ok(db);
+        let _db = PickleDb::new_bin(db_path.clone(), PickleDbDumpPolicy::AutoDump);
+        //return Ok(db);
```
Comment (joshuef) [inline src/utils.rs:47]: lets remove commented code

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**